scripts/results_verification.py

```diff
@@ -0,0 +1,74 @@
+import sys
+import os
+from pathlib import Path
+import xml.etree.ElementTree as ET
+
+
+def check_file_size(file_path, max_size=5):
+    """
+    Check if the file size of a given file is smaller than max_size.
+
+    :param file_path: path of junit xml report from sanitycheck to be cheched
+    :param max_size: max size allowed for the given file
+    """
+    file_size = file_path.stat().st_size / 1000000
+    return bool(file_size <= max_size)
+
+
+def check_item_count(file_path=None, item=None, max_count=None):
+    """
+    Check if the count of a given item is smaller than max_count. Item can be e.g. 'errors' or 'failures'
+
+    :param file_path: path of junit xml report from sanitycheck to be cheched
+    :param item: item to be verified
+    :param max_count: max amount of the given item
+    """
+    testsuite = ET.parse(file_path).getroot()
+    for summary in testsuite:
+        item_count = int(summary.attrib[item])
+        return bool(item_count <= max_count)
+
+
+def check_version(file_path=None, version=None):
+    """
+    Check if zephyr's version given in the report match the expected one
+
+    :param file_path: junit xml report from sanitycheck
+    :param version: expected version of zephyr
+    """
+    testsuite = ET.parse(file_path).getroot()[0]
+    properties = testsuite[0]
+    for property in properties:
+        if property.attrib['name'] == "version":
+            return bool(property.attrib['value'] == version)
+
+    print("Version not found.")
+    return False
+
+
+def main(path, version):
+    # TODO: add argparse
+    max_size = 5
+    max_failures = 50
+    max_errors = 50
```

Comment:
It seems reasonnable to add such limits.
Though, I'm affraid this could be a bit low when you have multiple boards on a bench.
I should have at least 15 boards on the bench (in good days).
If there are 4 tests failing on all these boards as this could happen for a kernel test that may impact all the boards in the same way I'm above this limit...
Or maybe there is a a way not to count multiple failures for the same test?

Reply:
Right now I see a workflow in a way that this script will be called for every file in a commit (one file=one platform). So it is actually a limit per platform. And of course, the numbers are now just from the top of my head. I will add argparsing so the numbers can be directly addressed when the script is called.

Reply:
Ok, sure if that's per boards it's reasonable!